pkg/user/tx_client.go

```diff
@@ -516,6 +517,47 @@ func (client *TxClient) EstimateGas(ctx context.Context, msgs []sdktypes.Msg, op
 	return client.estimateGas(ctx, txBuilder)
 }
 
+// EstimateGasPriceAndUsage simulates the transaction, calculating the amount of gas that was consumed during execution. The final
+// result will be multiplied by gasMultiplier(that is set in TxClient).
+// Also returns the estimated gas price given the provided priority.
+func (client *TxClient) EstimateGasPriceAndUsage(
+	ctx context.Context,
+	msgs []sdktypes.Msg,
+	priority gasestimation.TxPriority,
+	opts ...TxOption,
+) (float64, uint64, error) {
+	client.mtx.Lock()
+	defer client.mtx.Unlock()
+
+	txBuilder, err := client.signer.txBuilder(msgs, opts...)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	// add at least 1utia as fee to builder as it affects gas calculation.
+	txBuilder.SetFeeAmount(sdktypes.NewCoins(sdktypes.NewCoin(appconsts.BondDenom, sdktypes.NewInt(1))))
```

Comment:
Can this comment describe how the 1utia affects gas calculation? This seems kinda suspicious

Reply:
the estimation of a transaction that contains a specified fee is different than one that doesn't, and since it's a fixed field if I understand correctly, we add it here to have a more exact estimation.

Note: this was existing in the code, and left it based on the above understanding